deployment/ccip/deploy.go

```diff
@@ -60,6 +62,207 @@ var (
 	BurnMintTokenPool deployment.ContractType = "BurnMintTokenPool"
 )
 
+func DeployPrerequisiteChainContracts(e deployment.Environment, ab deployment.AddressBook, selectors []uint64) error {
+	state, err := LoadOnchainState(e)
+	if err != nil {
+		e.Logger.Errorw("Failed to load existing onchain state", "err")
+		return err
+	}
+	for _, sel := range selectors {
+		chain := e.Chains[sel]
+		err = DeployPrerequisiteContracts(e, ab, state, chain)
+		if err != nil {
+			return errors.Wrapf(err, "failed to deploy prerequisite contracts for chain %d", sel)
+		}
+	}
+	return nil
+}
+
+// DeployPrerequisiteContracts deploys the contracts that can be ported from previous CCIP version to the new one.
+// This is only required for staging and test environments where the contracts are not already deployed.
```

Comment:
Wouldn't it be required for new chains as well? Or are the old contracts all replaced after the initial migration?

Reply:
This is mainly designed thinking migration as an immediate goal. Are you referring to adding a completely newchain in 1.5 which is not existing in 1.6? I think in that case as we will have to call this function.